shell/platform/windows/keyboard_key_embedder_handler.cc

```diff
@@ -264,8 +264,13 @@ void KeyboardKeyEmbedderHandler::KeyboardHookImpl(
     pressingRecords_[physical_key] = eventual_logical_record;
   } else {
     auto record_iter = pressingRecords_.find(physical_key);
-    assert(record_iter != pressingRecords_.end());
-    pressingRecords_.erase(record_iter);
+    // Synthesized keyup events can be handled causing the
+    // key state to be released before the keyup event is
+    // received. Therefore, only erase the record if the
+    // key is currently recorded as pressed.
+    if (record_iter != pressingRecords_.end()) {
+      pressingRecords_.erase(record_iter);
+    }
```

Comment:
```suggestion
    } else {
      assert(record_iter != pressingRecords_.end());
    }
```

Reply:
Why have this? The condition will always be false just by virtue of the fact that in enters this else statement.

Reply:
We'd still like it to assert in debug mode, but prevent erase from  crashing by segfault in release mode since the error wouldn't be caught by assert. See https://discord.com/channels/608014603317936148/608020180177780791/1019033994286866433

Reply:
For that, I think `assert(false)` may be a better fit for clarity, as this condition will always be false in this branch.

Reply:
Are you certain? Would that assert not fail in exactly the same case that caused this issue in the first place? If we receive a keydown event while a key's state is not pressed, the synchronization will erase its pressing record before the next keyup event is processed, which will then reach and fail this `assert`. We would no longer crash in release mode, but I had thought our plan was to discard the redundant keyup events in these cases, not fail on them.